src/utils/is.js

```diff
@@ -36,3 +36,7 @@ export function isObjectLike ( thing ) {
 	const type = typeof thing;
 	if ( type === 'object' || type === 'function' ) return true;
 }
+
+export function isNotAnElement ( name ){
+	return document.createElement( name ).constructor === HTMLUnknownElement;
```

Comment:
I think that may be a bit too expensive to do for _every_ element that's created. I think #1723 is looking for a parse-time warning, which is nigh impossible if you want to be able to use components. I think the only thing that can be reasonably implemented from that issue is a strict(er) mode that requires closing tags and maybe has a few more smarts about what elements can be nested where.